vochain/scrutinizer/vote.go

```diff
@@ -59,7 +59,18 @@ func (s *Scrutinizer) GetEnvelope(nullifier []byte) (*models.VoteEnvelope, []byt
 	if envelope == nil {
 		return nil, nil, fmt.Errorf("transaction is not an Envelope")
 	}
-	return envelope, stx.Signature, nil
+	envelope.Nullifier = voteRef.Nullifier
+	return &types.EnvelopePackage{
+		Nonce:                envelope.Nonce,
+		ProcessId:            envelope.ProcessId,
+		VotePackage:          envelope.VotePackage,
+		Nullifier:            nullifier,
+		EncryptionKeyIndexes: envelope.EncryptionKeyIndexes,
+		Weight:               voteRef.Weight.Bytes(),
```

Comment:
Make weight it a big.Int and do `new(big.Int).SetBytes(voteRef.Weight.Bytes()))`

Reply:
Or just `new(big.Int).Set(voteRef.Weight)`, assuming you just want to make a copy.